examples/material_gallery/lib/demo/weathers_demo.dart

```diff
@@ -0,0 +1,495 @@
+                  ),
+                  new WeatherButton(
+                    onPressed: () {
+                      setState(() {
+                        weatherWorld.weatherType = WeatherType.snow;
+                      });
+                    },
+                    selected: weatherWorld.weatherType == WeatherType.snow,
+                    icon: "packages/flutter_gallery_assets/icon-snow.png"
+                  )
+                ]
+              )
+            )
+          ]
+        )
+      )
+    );
+  }
+}
+
+const double _kWeatherButtonSize = 56.0;
+const double _kWeatherIconSize = 36.0;
+
+class WeatherButton extends StatelessComponent {
+  WeatherButton({this.icon, this.selected, this.onPressed});
+
+  final String icon;
+  final bool selected;
+  VoidCallback onPressed;
+
+  Widget build(BuildContext context) {
+    Color color;
+    if (selected)
+      color = Theme.of(context).primaryColor;
+    else
+      color = new Color(0x33000000);
+
+    return new Padding(
+      padding: new EdgeDims.all(15.0),
+      child: new Material(
+        color: color,
+        type: MaterialType.circle,
+        elevation: 0,
+        child: new Container(
+          width: _kWeatherButtonSize,
+          height: _kWeatherButtonSize,
+          child: new InkWell(
+            onTap: onPressed,
+            child: new Center(
+              child: new AssetImage(
+                name: icon,
+                width: _kWeatherIconSize,
+                height: _kWeatherIconSize
+              )
+            )
+          )
+        )
+      )
+    );
+  }
+}
+
+List<Color> _kBackgroundColorsTop = [
+  new Color(0xff5ebbd5),
+  new Color(0xff0b2734),
+  new Color(0xffcbced7)
+];
+
+List<Color> _kBackgroundColorsBottom = [
+  new Color(0xff4aaafb),
+  new Color(0xff4c5471),
+  new Color(0xffe0e3ec)
+];
+
+class WeatherWorld extends NodeWithSize {
+  WeatherWorld() : super(new Size(2048.0, 2048.0)) {
+    _background = new GradientNode(
+      this.size,
+      _kBackgroundColorsTop[0],
+      _kBackgroundColorsBottom[0]
+    );
+    addChild(_background);
+
+    _cloudsSharp = new CloudLayer(_images['packages/flutter_gallery_assets/clouds-0.png'], false, false, 20.0);
+    addChild(_cloudsSharp);
+
+    _cloudsDark = new CloudLayer(_images['packages/flutter_gallery_assets/clouds-1.png'], true, true, 40.0);
+    addChild(_cloudsDark);
+
+    _cloudsSoft = new CloudLayer(_images['packages/flutter_gallery_assets/clouds-1.png'], false, false, 60.0);
+    addChild(_cloudsSoft);
+
+    _sun = new Sun();
+    _sun.position = new Point(1024.0, 1024.0);
+    _sun.scale = 1.5;
+    addChild(_sun);
+
+    _rain = new Rain();
+    addChild(_rain);
+
+    _snow = new Snow();
+    addChild(_snow);
+  }
+
+  GradientNode _background;
+  CloudLayer _cloudsSharp;
+  CloudLayer _cloudsSoft;
+  CloudLayer _cloudsDark;
+  Sun _sun;
+  Rain _rain;
+  Snow _snow;
+
+  WeatherType _weatherType = WeatherType.sun;
+
+  WeatherType get weatherType => _weatherType;
```

Comment:
put the getter on the line before the private field backing it. This makes it cleaner when you add dartdocs.